packages/mobile/src/send/useInviteRewardLimitDialog.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useSelector } from 'react-redux'
+import Dialog from 'src/components/Dialog'
+import { MAX_NUM_INVITES_WITH_REWARDS } from 'src/config'
+import { getAddressFromPhoneNumber } from 'src/identity/contactMapping'
+import {
+  e164NumberToAddressSelector,
+  secureSendPhoneNumberMappingSelector,
+} from 'src/identity/selectors'
+import { Recipient } from 'src/recipients/recipient'
+import { inviteRewardsActiveSelector, invitesWithRewardsSentSelector } from 'src/send/selectors'
+
+const useInviteRewardLimitDialog = (onNavigateSendAmount: (recipient: Recipient) => void) => {
+  const [invitee, setInvitee] = useState<Recipient | null>(null)
+
+  const invitesWithRewardsSent = useSelector(invitesWithRewardsSentSelector)
+  const inviteRewardsEnabled = useSelector(inviteRewardsActiveSelector)
+  const secureSendPhoneNumberMapping = useSelector(secureSendPhoneNumberMappingSelector)
+  const e164NumberToAddress = useSelector(e164NumberToAddressSelector)
+
+  const { t } = useTranslation()
+
+  const handleClose = () => {
+    setInvitee(null)
+  }
+
+  const handleInvite = () => {
+    if (invitee) {
+      onNavigateSendAmount(invitee)
+      setInvitee(null)
+    }
+  }
+
+  const onProceedSelectRecipient = (recipient: Recipient) => {
+    if (inviteRewardsEnabled && !recipient.address && recipient.e164PhoneNumber) {
+      const recipientAddress = getAddressFromPhoneNumber(
```

Comment:
i will find another way to do this, i realised that at this point the user hasn't gone through secure send yet ☹️